### PR TITLE
WSTEAMA-989: Upgrade glob to 10.3.10

### PR DESCRIPTION
**Pull request by karinathomasbbc**
Resolves JIRA https://jira.dev.bbc.co.uk/browse/WSTEAMA-989

Overall changes
======

Code changes
======
- Major upgrade of glob from 8 -> 10

Testing
======
glob is currently used for preloading fonts in storybook
[Storybook Permalink](https://wsteama-989-upgrade-glob--5d28eb3fe163f6002046d6fa.chromatic.com/?path=/story/docs-a11y-testing-cross-device--page)
Check the HTML of storybook, and ensure that all fonts are declared as `rel=preload`

e.g.

Screenshot 2024-02-12 at 16 34 48

### CODE REVIEW

File: dependencyFreshnessTable.csv
Comment (karinathomasbbc) [inline dependencyFreshnessTable.csv:1]: This file was renamed from `depenencyFreshnessTable`, but has also been sorted alphabetically

File: esmDependencyTable.csv
Comment (karinathomasbbc) [inline esmDependencyTable.csv:1]: This file has been updated to sort dependency name alphabetically, as a result of running `yarn esmDependencyCheck`

File: scripts/esmDependencyCheck.js
Comment (karinathomasbbc) [inline scripts/esmDependencyCheck.js:1]: Fix linting issues & update sorting to be by name only

File: scripts/dependencyFreshness.js
Comment (karinathomasbbc) [inline scripts/dependencyFreshness.js:1]: Fixed linting issues & sorting alphabetically by dependency name

**Review by amoore108 — APPROVED** [review]
Seems grand, may be worth testing on Opera Mini Extreme to make sure it works correctly?

**Comment by karinathomasbbc** [comment]
> Seems grand, may be worth testing on Opera Mini Extreme to make sure it works correctly?

Do you mean test Storybook on Opera Mini Extreme? Or Simorgh?

**Comment by amoore108** [comment]
> > Seems grand, may be worth testing on Opera Mini Extreme to make sure it works correctly?
> 
> Do you mean test Storybook on Opera Mini Extreme? Or Simorgh?

Simorgh probably to be the most accurate. Its tricky to know for sure until its "out in the wild" unfortunately. Just the nature of Opera Mini, it looks like it may work until its actually ran on the device.

**Comment by karinathomasbbc** [comment]
> > > Seems grand, may be worth testing on Opera Mini Extreme to make sure it works correctly?
> > 
> > 
> > Do you mean test Storybook on Opera Mini Extreme? Or Simorgh?
> 
> Simorgh probably to be the most accurate. Its tricky to know for sure until its "out in the wild" unfortunately. Just the nature of Opera Mini, it looks like it may work until its actually ran on the device.

Ah yes OK, that's fair. Glob is used by other dependencies, so you never know how it might affect Opera Mini.. Which also reminds me... we should probably try and write some form of test / script which will allow us to detect potential issues with Opera Mini.. not for now. 

Next question... how do we test with Opera Mini Extreme - Saucelabs?

**Comment by amoore108** [comment]
> > > > Seems grand, may be worth testing on Opera Mini Extreme to make sure it works correctly?
> > > 
> > > 
> > > Do you mean test Storybook on Opera Mini Extreme? Or Simorgh?
> > 
> > 
> > Simorgh probably to be the most accurate. Its tricky to know for sure until its "out in the wild" unfortunately. Just the nature of Opera Mini, it looks like it may work until its actually ran on the device.
> 
> Ah yes OK, that's fair. Glob is used by other dependencies, so you never know how it might affect Opera Mini.. Which also reminds me... we should probably try and write some form of test / script which will allow us to detect potential issues with Opera Mini.. not for now.
> 
> Next question... how do we test with Opera Mini Extreme - Saucelabs?

Yea we're looking into ways to help make Opera Mini more resilient https://jira.dev.bbc.co.uk/browse/NEWSWORLDSERVICE-1797

We tend to test on physical devices. Opera Mini Extreme can be enabled on Opera Mini on Android, so if you have Android phone you should be able to try it out. Annoyingly, the Android version is most likely _not_ the one that is used in reality in the Hausa service. They typically use feature phones, so run a very old version of Opera.

**Review by shayneahchoon — APPROVED** [review]

**Pull request merged**